File: 4um-backend/src/main/java/com/github/boardyb/forum/discussion/Discussion.java
package com.github.boardyb.forum.discussion;

import lombok.Data;
import org.springframework.data.annotation.CreatedDate;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.validation.constraints.NotBlank;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "discussions", schema = "forum")
public class Discussion {

    @Id
    @Column(name = "id")
    private String id;
    @Column(name = "post_count")
    private Integer postCount;
    @Column(name = "creation_date")
    @CreatedDate
    private LocalDateTime creationDate;
    @NotBlank
    @Column(name = "title")
    private String title;
    @Column(name = "description")
    private String description;
    @Column(name = "creator")
    private String creator;
    @Column(name = "featured")
    private boolean featured = false;
    @Column(name = "deleted")
    private boolean deleted = false;

}

File: 4um-frontend/src/app/discussion/discussion.ts
import { Entity } from "../core/repository/entity";

export class Discussion extends Entity<string> {
  postCount: number;
  creationDate: Date;
  title: string;
  description: string;
  creator: string;
  featured: boolean;
  deleted: boolean;

  public static deserialize(obj: any): Discussion {
    const discussion = this.createEmptyDiscussion();
    discussion.setId(obj.id);
    discussion.postCount = obj.postCount;
    discussion.creationDate = obj.creationDate;
    discussion.title = obj.title;
    discussion.description = obj.description;
    discussion.creator = obj.creator;
    discussion.featured = obj.featured;
    discussion.deleted = obj.deleted;
    return discussion;
  }

  public static createEmptyDiscussion(): Discussion {
    return new Discussion(null, null, null, null, new Date(), null, false, false);
  }


  constructor(id: string, postCount: number, title: string, description: string, creationDate: Date, creator: string, isFeatured: boolean, isDeleted: boolean) {
    super();
    this.setId(id);
    this.postCount = postCount;
    this.creationDate = creationDate;
    this.creator = creator;
    this.featured = isFeatured;
    this.deleted = isDeleted;
  }
}
